### ISSUE TITLE
x on parenthesis not working in some cases

### ISSUE THREAD

**Issue by justbur** [luxbock/evil-cleverparens#38]
Test case

``` elisp
(["]a" "b")
```

I'm not sure what's going on here, and I don't have time to test a bunch of cases right now, but if you hit x from the state above you get this which is not what I expected.

``` elisp
("[a]" "b")
```

**Comment by sooheon** [comment]
Can confirm I'm seeing this as well. Deleting opening quotes does not splice the sexp.

**Comment by sooheon** [comment]
Additionally, `x` on the closing quote just imbalances the quotes:

```
("a["] "b")

("a[ ]"b")
```

**Comment by luxbock** [comment]
I've looked at this a little bit and I think it might be a bug in a function which needs step-through debugging to fix, and I haven't had much time to dive into it. I'll give it another go later.

I've also noticed some weird behavior with strings in other places as well, which I think are due to `smartparens`, so I'll try to make a note of when that happens again and submit an issue upstream. I think these two are a problem with `evil-cleverparens` however.
